nx.json

```diff
@@ -6,12 +6,24 @@
         "cacheableOperations": [
           "build:bundle",
           "build:transpile",
-          "build:tarball",
           "build:types"
         ]
       }
     }
   },
+  "namedInputs": {
+    "sharedGlobals": [
+      "{workspaceRoot}/*.js",
+      "{workspaceRoot}/*.json",
+      "{workspaceRoot}/yarn.lock"
+    ],
+    "default": [
```

Comment:
note on this: This means that by default, a job is considered changed when one of these given files changes. meaning we do not consider a build to have to re-run when we change only a file in the tests, or a markdown file.